contracts/Lens/JoeLens.sol

```diff
@@ -255,6 +273,21 @@ contract JoeLens is Exponential {
         }
     }
 
+    function _setRewardDistributor(address payable newRewardDistributor) public returns(uint256) {
+        if (msg.sender != admin) {
+            return uint256(Error.UNAUTHORIZED);
+        }
+
+        (bool success, ) = newRewardDistributor.call.value(0)(abi.encodeWithSignature("initialize()", 0));
+        if (!success) {
+            return uint256(Error.REJECTION);
+        }
+
+        rewardDistributor = newRewardDistributor;
+
+        return uint256(Error.NO_ERROR);
+    }
+
```

Comment:
Ok 👍 . I took this function logic from the ` _setRewardDistributor` function in `Joetroller`, could you explain why the initialize call was needed in that contract?

Reply:
It's needed in Joetroller because Joetroller and RewardDistributor are linked. Users interact with Joetroller to claim rewards, which then interacts with RewardDistributor to get those rewards and return it to the users.

In this case, you're just interacting with JoeLens, which is purely returning informational data.